legacy/python/platform/src/python_platform.rs

```diff
@@ -0,0 +1,311 @@
+        let tool = self.toolchain.get_for_version(&req)?;
+
+        Ok(Box::new(tool))
+    }
+
+    fn get_dependency_configs(&self) -> miette::Result<Option<(String, String)>> {
+        Ok(Some((
+            "requirements.txt".to_owned(),
+            "requirements.txt".to_owned(),
+        )))
+    }
+
+    async fn setup_toolchain(&mut self) -> miette::Result<()> {
+        let req = match &self.config.version {
+            Some(v) => RuntimeReq::Toolchain(v.to_owned()),
+            None => RuntimeReq::Global,
+        };
+
+        let mut last_versions = FxHashMap::default();
+
+        if !self.toolchain.has(&req) {
+            self.toolchain.register(
+                &req,
+                PythonTool::new(
+                    Arc::clone(&self.proto_env),
+                    Arc::clone(&self.console),
+                    &self.config,
+                    &req,
+                )
+                .await?,
+            );
+        }
+
+        self.toolchain.setup(&req, &mut last_versions).await?;
+
+        Ok(())
+    }
+
+    async fn teardown_toolchain(&mut self) -> miette::Result<()> {
+        self.toolchain.teardown_all().await?;
+
+        Ok(())
+    }
+
+    // ACTIONS
+
+    #[instrument(skip_all)]
+    async fn setup_tool(
+        &mut self,
+        _context: &ActionContext,
+        runtime: &Runtime,
+        last_versions: &mut FxHashMap<String, UnresolvedVersionSpec>,
+    ) -> miette::Result<u8> {
+        let req = &runtime.requirement;
+
+        if !self.toolchain.has(req) {
+            self.toolchain.register(
+                req,
+                PythonTool::new(
+                    Arc::clone(&self.proto_env),
+                    Arc::clone(&self.console),
+                    &self.config,
+                    req,
+                )
+                .await?,
+            );
+        }
+
+        let installed = self.toolchain.setup(req, last_versions).await?;
+
+        Ok(installed)
+    }
+
+    #[instrument(skip_all)]
+    async fn install_deps(
+        &self,
+        _context: &ActionContext,
+        runtime: &Runtime,
+        working_dir: &Path,
+    ) -> miette::Result<Vec<Operation>> {
+        actions::install_deps(
+            self.toolchain.get_for_version(&runtime.requirement)?,
+            self.workspace_root.as_path(),
+            working_dir,
+            &self.console,
+        )
+        .await
+    }
+
+    #[instrument(skip_all)]
+    async fn sync_project(
+        &self,
+        _context: &ActionContext,
+        _project: &Project,
+        _dependencies: &FxHashMap<Id, Arc<Project>>,
+    ) -> miette::Result<bool> {
+        Ok(false)
+    }
+
+    #[instrument(skip_all)]
+    async fn hash_manifest_deps(
+        &self,
+        manifest_path: &Path,
+        hasher: &mut ContentHasher,
+        _hasher_config: &HasherConfig,
+    ) -> miette::Result<()> {
+        if let Some(python_version) = &self.config.version {
```

Comment:
We should probably still hast he dependencies, even if no version is configured.

Reply:
Okay